src/modal-checkout/modal.js

```diff
@@ -49,40 +49,53 @@ domReady( () => {
 
 	const initialHeight = modalContent.clientHeight + spinner.clientHeight + 'px';
 	const closeCheckout = () => {
-		spinner.style.display = 'flex';
-
 		const container = iframe?.contentDocument?.querySelector( `#${ IFRAME_CONTAINER_ID }` );
-
-		if ( iframe && modalContent.contains( iframe ) ) {
-			// Reset iframe and modal content heights.
-			iframe.src = 'about:blank';
-			iframe.style.height = '0';
-			modalContent.removeChild( iframe );
-			modalContent.style.height = initialHeight;
+		const afterSuccessUrlInput = container.querySelector( 'input[name="after_success_url"]' );
+		const afterSuccessBehaviorInput = container.querySelector(
+			'input[name="after_success_behavior"]'
+		);
+		const hasNewsletterPopup = document?.querySelector( '.newspack-newsletters-signup-modal' );
+
+		// We want to block closing the modal if redirecting elsewhere:
+		let shouldCloseModal = true;
```

Comment:
Seems like we can just define this directly using the various checks for after success params:

```suggestion
		const shouldCloseModal = ! afterSuccessUrlInput && ! afterSuccessBehaviorInput;
```

Reply:
Thanks @chickenn00dle! The only issue I found with this specific change is that, without the `container?.checkoutComplete` check, it could block the modal from being closed by the 'Close' button, too. 

I made a modified version of this feedback in [38b6d13](https://github.com/Automattic/newspack-blocks/pull/1855/commits/38b6d13fa73348a78637420dd280455d66215320) -- could I please get your eyes on this one more time to make sure I'm not missing something?